packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -718,12 +719,33 @@ mixin SchedulerBinding on BindingBase, ServicesBinding {
 
   bool _warmUpFrame = false;
 
+  void _warmUpSkiaShaderCompilations({ShaderWarmUp customShaderWarmUp}) {
+    final PictureRecorder recorder = PictureRecorder();
+    final Canvas canvas = Canvas(recorder);
+
+    if (customShaderWarmUp != null) {
+      customShaderWarmUp(canvas);
+    } else {
+      defaultShaderWarmUp(canvas);
```

Comment:
You could just make `defaultShaderWarmUp` the default value of `customShaderWarmUp`.

Reply:
I discussed with @zanderso and it seems that Dart doesn't support default value of non-const variable (`defaultShaderWarmUp` is not const).

Reply:
This code prints "foo":

```dart
void foo() { print('foo'); }

void bar([ void	Function() arg = foo ]) { arg(); }
      
void main() {
  bar();
}
```